### PR TITLE
Design principles for development on AMP

### PR DESCRIPTION
**Pull request by cramforce**
The goal is to define a set of principles or guidelines that help us make more consistent decisions as we continue develpment on AMP.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -43,18 +43,19 @@ Code contributors to the AMP HTML project must sign a Contributor License Agreem
 ### Contributing Features
 
 All pull requests for new features must go through the following process:
-* Intent-to-implement GitHub issue started for discussion
+* Please damiliarize yourself with the [AMP Design Principles](DESIGN_PRINCIPLES).md
```
Comment (erwinmombay) [inline CONTRIBUTING.md]: damiliarize -> familiarize

Comment (cramforce) [inline CONTRIBUTING.md]: Fixed

File: DESIGN_PRINCIPLES.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+We’ve chosen to build AMP as a library in the spirit of the [extensible web manifesto](https://github.com/extensibleweb/manifesto/blob/master/README.md) to be able to fix the web of today, not the web of tomorrow.
+
+AMP should be fast in today's browsers. When certain optimizations aren't possible with today's platform, AMP developers should participate in standards development to get these added to the web platform.
+
+## Don’t break the web.
+
+Ensure that if AMP has outages or problems it doesn’t hurt the rest of the web. That means if the Google AMP Cache, the URL API or the library fails it should be possible for websites and consumption apps to gracefully degrade. If something works with an AMP cache it should also work without a cache.
+
+
+## User Experience > Developer Experience > ease of implementation.
```
Comment (aghassemi) [inline DESIGN_PRINCIPLES.md]: Ease of Implementation for consistent letter casing.

File: DESIGN_PRINCIPLES.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+When in doubt, do what’s best for the end user experience, even if it means that it’s harder for the page creator to build or for the library developer to implement.
```
Comment (lannka) [inline DESIGN_PRINCIPLES.md]: Shall we swap position of this item with the "Don’t design for a hypothetical faster future browser."
I think this one is higher level and very important.

File: DESIGN_PRINCIPLES.md (hunk 2 above)
Comment (cramforce) [inline DESIGN_PRINCIPLES.md]: Done

File: DESIGN_PRINCIPLES.md (hunk 3 above)
Comment (cramforce) [inline DESIGN_PRINCIPLES.md]: Moved it up. Great point!

**Comment by lannka** [comment]
This is great. LGTM

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -43,18 +43,19 @@ Code contributors to the AMP HTML project must sign a Contributor License Agreem
 ### Contributing Features
 
 All pull requests for new features must go through the following process:
-* Intent-to-implement GitHub issue started for discussion
+* Please familiarize yourself with the [AMP Design Principles](DESIGN_PRINCIPLES).md
+* Start and Intent-to-implement GitHub issue for discussion of the new feature.
```
Comment (jridgewell) [inline CONTRIBUTING.md]: "an"

**Comment by aghassemi** [comment]
LGTM

**Comment by adewale** [comment]
LGTM

**Comment by dvoytenko** [comment]
LGTM

**Pull request merged**